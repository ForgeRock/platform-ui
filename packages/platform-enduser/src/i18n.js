/**
 * Copyright (c) 2020-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { createI18n } from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

export default createI18n({
  messages: loadLocaleMessages(),
  silentFallbackWarn: true,
  silentTranslationWarn: true,
  dateTimeFormats,
});
