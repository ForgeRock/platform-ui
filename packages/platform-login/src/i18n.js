/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Vue from 'vue';
import VueI18n from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

Vue.mixin({
  methods: {
    translationExists(path) {
      return this.$t(path) !== path;
    },
  },
});

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
