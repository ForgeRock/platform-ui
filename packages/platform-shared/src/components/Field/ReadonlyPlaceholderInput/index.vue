<!-- Copyright (c) 2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <FrBasicInput
    v-bind="$attrs"
    type="string"
    :value="valueToDisplay"
    readonly>
    <template
      #append
      v-if="showClearField">
      <BInputGroupAppend>
        <BButton
          @click="clearField"
          class="clear-btn"
          :aria-label="$t('common.clear')">
          <FrIcon
            name="close"
          />
        </BButton>
      </BInputGroupAppend>
    </template>
  </FrBasicInput>
</template>

<script>
import { getPlaceholderValueToDisplay } from '@forgerock/platform-shared/src/utils/esvUtils';
import FrBasicInput from '@forgerock/platform-shared/src/components/Field/BasicInput';
import { BButton, BInputGroupAppend } from 'bootstrap-vue';
import FrIcon from '@forgerock/platform-shared/src/components/Icon';

/**
 * Input for displaying ESV placeholder values. The values are shown as a readonly string, with a button to clear the placeholder so a new value can be set.
 */
export default {
  name: 'ReadonlyPlaceholderInput',
  components: {
    FrBasicInput,
    FrIcon,
    BButton,
    BInputGroupAppend,
  },
  props: {
    value: {
      type: [Array, Object, Number, String, Boolean],
      default: '',
    },
  },
  inject: {
    showClearField: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return {
      valueToDisplay: getPlaceholderValueToDisplay(this.value),
    };
  },
  methods: {
    /**
    * Resets the field and clears its placeholder value
    */
    clearField() {
      this.$emit('input', '');
    },
  },
};
</script>

<style lang="scss" scoped>
  .clear-btn {
    border-left: none;
    background-color: $gray-100 !important;
  }

  .form-label-group:focus-within {
   .input-buttons:not(:focus-within) .input-group-append .btn {
        border-color: $primary !important;
        clip-path: inset(-1px -1px -1px 0px) !important;
        box-shadow: 0 0 0 0.0625rem $primary !important;
   }
 }
</style>
