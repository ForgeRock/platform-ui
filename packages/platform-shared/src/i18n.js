/**
 * Copyright 2019 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import dateTimeFormats from './dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
