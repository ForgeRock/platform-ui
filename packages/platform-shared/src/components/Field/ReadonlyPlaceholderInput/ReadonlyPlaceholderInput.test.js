/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { mount } from '@vue/test-utils';
import ValidationRules from '@forgerock/platform-shared/src/utils/validationRules';
import ReadonlyPlaceholderInput from './index';
import i18n from '@/i18n';

ValidationRules.extendRules({
  required: ValidationRules.getRules(i18n).required,
});

const defaultProps = {
  name: 'name',
  autofocus: false,
  validation: 'required',
  'validation-immediate': true,
};

describe('ReadonlyPlaceholderInput', () => {
  function setup(props, provide) {
    return mount(ReadonlyPlaceholderInput, {
      global: {
        plugins: [i18n],
        provide,
      },
      attachTo: document.body,
      props: {
        ...defaultProps,
        ...props,
      },
    });
  }

  describe('@renders', () => {
    it('shows a string placeholder value as-is in a readonly text input', async () => {
      const wrapper = setup({ value: '&{esv-test}' });
      await wrapper.vm.$nextTick();

      const input = wrapper.find('input');
      expect(input.attributes('readonly')).toBe('');
      expect(input.element.value).toBe('&{esv-test}');
    });

    it('shows the placeholder key from an object value in a readonly text input', async () => {
      const wrapper = setup({ value: { $string: '&{esv-fromobject}' } });
      await wrapper.vm.$nextTick();

      const input = wrapper.find('input');
      expect(input.attributes('readonly')).toBe('');
      expect(input.element.value).toBe('&{esv-fromobject}');
    });

    it('Adds text-truncate class to input when in a readonly text input', async () => {
      const wrapper = setup({ value: '&{esv-test}' });
      const input = wrapper.find('input');
      expect(input.classes().includes('text-truncate')).toBeTruthy();
    });

    it('shows a clear button on the field', async () => {
      const wrapper = setup({ value: '&{esv-test}' });
      await wrapper.vm.$nextTick();

      const clearButton = wrapper.find('button').element;
      expect(clearButton).toBeVisible();
    });

    it('hides clear button on the field when showClearField set to false', async () => {
      const wrapper = setup({ value: '&{esv-test}' }, { showClearField: false });
      await wrapper.vm.$nextTick();

      const clearButton = wrapper.find('button');
      expect(clearButton.exists()).toBeFalsy();
    });
  });

  describe('@actions', () => {
    it('should clear the input value when the clear button is clicked', async () => {
      const wrapper = setup({ value: '&{esv-test}' });
      await wrapper.vm.$nextTick();

      const clearButton = wrapper.find('button');
      clearButton.trigger('click');
      expect(wrapper.emitted().input).toStrictEqual([['']]);
    });
  });
});
