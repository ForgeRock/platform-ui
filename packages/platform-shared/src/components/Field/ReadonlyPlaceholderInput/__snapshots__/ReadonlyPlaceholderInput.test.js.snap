// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadonlyPlaceholderInput @renders shows a string placeholder value as-is in a readonly text input: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (2)",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) Functional component <BInputGroupAppend> should be defined as a plain function in Vue 3. The "functional" option has been removed. NOTE: Before migrating to use plain functions for functional components, first make sure that all async components usage have been migrated and its compat behavior has been disabled.
  Details: https://v3-migration.vuejs.org/breaking-changes/functional-components.html",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "placeholder" with v-bind value \`false\` will render placeholder="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "aria-describedby" with v-bind value \`false\` will render aria-describedby="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <BInputGroupAppend>",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (3)",
    "
",
    " at <BInputGroupAppend>",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) (2)",
    "
",
    " at <BInputGroupAppend>",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (2)",
    "
",
    " at <BInputGroupAddon",
    "id=undefined",
    "isText=false",
    "tag="div"",
    " ...",
    ">",
    "
",
    " at <BInputGroupAppend>",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (3)",
    "
",
    " at <BButton",
    "onClick=fn<bound clearField>",
    "class="clear-btn"",
    "aria-label="Clear"",
    ">",
    "
",
    " at <BInputGroupAddon",
    "id=undefined",
    "isText=false",
    "tag="div"",
    " ...",
    ">",
    "
",
    " at <BInputGroupAppend>",
    "
",
    " at <InputLayout",
    "id="floatingLabelInput4"",
    "name="name"",
    "description=""",
    " ...",
    ">",
    "
",
    " at <BasicInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <ReadonlyPlaceholderInput",
    "name="name"",
    "autofocus=false",
    "validation="required"",
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
